src/structures/idt.rs

```diff
@@ -593,17 +604,41 @@ impl<T> PartialEq for Entry<T> {
 }
 
 /// A handler function for an interrupt or an exception without error code.
+#[cfg(feature = "abi_x86_interrupt")]
 pub type HandlerFunc = extern "x86-interrupt" fn(InterruptStackFrame);
+#[cfg(not(feature = "abi_x86_interrupt"))]
+#[derive(Copy, Clone, Debug)]
+pub struct HandlerFunc(());
```

Comment:
While it's true that the docs on docs.rs don't show this documentation, we should still put something here for people that use `cargo doc` to view the docs locally. A short sentence like the following would be enough:

```suggestion
/// This type is not usable without the `abi_x86_interrupt` feature.
#[cfg(not(feature = "abi_x86_interrupt"))]
#[derive(Copy, Clone, Debug)]
pub struct HandlerFunc(());
```

(This also applies to the other type aliases.)